v2/errors.go

```diff
@@ -38,7 +38,17 @@ type HTTPStatusCodeError struct {
 }
 
 func (e HTTPStatusCodeError) Error() string {
-	return fmt.Sprintf("Status: %v; ErrorMessage: %v; Description: %v; ResponseError: %v", e.StatusCode, e.ErrorMessage, e.Description, e.ResponseError)
+	if e.ErrorMessage == nil {
```

Comment:
Nevermind, I am an idiot.  If you revise this to not mutate the error, I would be fine with it.